_posts/2017-12-21-totw-42.md

```diff
@@ -0,0 +1,92 @@
+---
+title: "Tip of the Week #42: Prefer Factory Functions to Initializer Methods"
+layout: tips
+sidenav: side-nav-tips.html
+published: false
+permalink: tips/42
+type: markdown
+order: "042"
+---
+
+Originally posted as totw/42 on 2013-05-10
+
+By Geoffrey Romer [(gromer@google.com)](mailto:gromer@google.com)
+
+Revised 2017-12-21
+
+*"The man who builds a factory builds a temple; the man who works there worships
+there, and to each is due, not scorn and blame, but reverence and praise." --
+Calvin Coolidge*
+
+One of the main difficulties with writing constructors in C++ (at Google) is
+that they effectively must succeed, because they have no way to report failure
+to the caller. (Google's C++ style forbids exceptions.) You can use an
```

Comment:
I sorta don't like this phrasing - I've almost never actually cared about the fact that our c'tors don't throw. It certainly doesn't rise to the level of "difficulty".  

Maybe something like:
In environments where exceptions are disallowed, constructors must succeed, because they have no way to report failure to the caller.

Reply:
Rewritten. I would say that is it is a "difficulty" only in that it requires an extra bit of care, but PTAL.